### PR TITLE
Add Aviris NG Collection to Aviris Catalog

### LINKED ISSUE #79: Add AVIRIS-ng to Franklin
AVIRIS and AVIRIS-ng are cataloged separately by JPL/NASA.  The latter has more international flights, and half the FWHM of AVIRIS classic (and zero calories!).  The source table is here: https://docs.google.com/spreadsheets/d/1v6VoBodTqPdhAZ9Goc4vhpwVLew8KGfYyLJDqb151pI/edit#gid=1403228846

I can confirm that all our existing tools should work for this, but the FTP server seems a bit touchier.  Beware.  We may want to add a `is_ng` property?  Or just add this to a separate table?  Better's choice.

### LINKED ISSUE #93: Use layers extension in generated STAC catalogs
At this point we have multiple collections. An each collection represents a separate `layer`. 
However, I'd like to generalize collections a bit further. 
This task is to add `layer:ids` property into all generated items according to the [Layer STAC Extension spec](https://github.com/azavea/stac4s/tree/master/docs/stac-spec/extensions/layer).

Its usage example is shown below:
```javascript
{
    "id" : "id",
    "stac_extensions" : ["layer"],
    // ...
    "properties": {
        "collection": "item-collection",
        "layer:ids" : ["layer-us", "layer-pa"]
    }
    // ...
}
```

It may seem a bit preamture, but it can simplify visualization of the STAC catalog #60

### PR DESCRIPTION
**Pull request by CloudNiner**
## Overview

The Aviris catalog now looks like:

- Catalog: "aviris"
  - Collection: "aviris-classic"
  - Collection: "aviris-ng"

Closes #79, Closes #93

## Notes

I refactored the one file script out to a few different files. Hopefully that keeps things a little be more logically separated. 

This PR is likely best digested by individual commit. The changes necessary to ingest NG are in the first commit. The refactor which does not affect functionality is in the second two.

## Testing Instructions

Run `./scripts/update` to ensure the catalogs-aviris container is up to date.

`docker-compose run --rm aviris-catalog` should now generate a catalog with both sub collections. This catalog should continue to successfully ingest into franklin.

## Checklist

- [x] `fixup!` commits have been squashed
- [x] CI passes after rebase
- [x] README.md updated if necessary to reflect the changes

### CODE REVIEW

**Comment by CloudNiner** [comment]
Once this is approved and merged I'll reupload the catalog to S3 and the hosted franklin instance.

File: src/catalogs/aviris/aviris_df/__init__.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+from functools import partial
+from pathlib import Path
+
+from geopandas import GeoDataFrame
+import pandas as pd
+
+from .converters import aviris_series_to_item
+
+
+AVIRIS_DESCRIPTION = "AVIRIS is an acronym for the Airborne Visible InfraRed Imaging Spectrometer. AVIRIS is a premier instrument in the realm of Earth Remote Sensing. It is a unique optical sensor that delivers calibrated images of the upwelling spectral radiance in 224 contiguous spectral channels (also called bands) with wavelengths from 400 to 2500 nanometers (nm). AVIRIS has been flown on four aircraft platforms: NASA's ER-2 jet, Twin Otter International's turboprop, Scaled Composites' Proteus, and NASA's WB-57. The ER-2 flies at approximately 20 km above sea level, at about 730 km/hr. The Twin Otter aircraft flies at 4km above ground level at 130km/hr. AVIRIS has flown all across the US, plus Canada and Europe. This catalog contains all AVIRIS missions from 2006 - 2019."
```
Comment (pomadchin) [inline src/catalogs/aviris/aviris_df/__init__.py:11]: Why not to use the [namespace package](https://docs.python.org/3/glossary.html#term-namespace-package) here? (just a curiosity question)

File: src/catalogs/aviris/main.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+from datetime import datetime, timezone
+
+import pystac
+import stacframes
+
+from aviris_df import AvirisClassic, AvirisNg, AVIRIS_DESCRIPTION
+
+
+def main():
```
Comment (pomadchin) [inline src/catalogs/aviris/main.py:9]: sweet!

**Review by pomadchin — COMMENTED** [review]
Perfectly works and the code looks nice! Thanks!

Btw, it looks like we would reingest both catalogs into franklin? 
Would you have time to add https://github.com/azavea/nasa-hyperspectral/issues/93 since we'll have to reingest everything anyway? 

TLDR; to add `stac_extensions: [layers]` into STAC Items and `properties.layer:ids`:

```javascript
{
    "id" : "id",
    "stac_extensions" : ["layer"],
    // ...
    "properties": {
        "collection": "aviris-classic",
        "layer:ids" : ["layer-aviris-classic"]
    }
    // ...
}
```

If you won't have time lets merge it as is.

**Review by pomadchin — APPROVED** [review]
Perfectly works and the code looks nice! Thanks!

Btw, it looks like we would reingest both catalogs into franklin? 
Would you have time to add https://github.com/azavea/nasa-hyperspectral/issues/93 since we'll have to reingest everything anyway? 

TLDR; to add `stac_extensions: [layers]` into STAC Items and `properties.layer:ids`:

```javascript
{
    "id" : "id",
    "stac_extensions" : ["layer"],
    // ...
    "properties": {
        "collection": "aviris-classic",
        "layer:ids" : ["layer-aviris-classic"]
    }
    // ...
}
```

If you won't have time lets merge it as is.

**Comment by CloudNiner** [comment]
@pomadchin 

> Would you have time to add #93

Added in https://github.com/azavea/nasa-hyperspectral/pull/89/commits/5bcba53c61b979966e1569e35b75fcce98a3f1a3

**Review by pomadchin — APPROVED** [review]
LGTM, thanks!
